projects/plugins/jetpack/extensions/blocks/slideshow/edit.js

```diff
@@ -192,14 +192,35 @@ export const SlideshowEdit = ( {
 	);
 };
 
+const memoCache = new Map();
```

Comment:
I'm wondering whether we should clear the map somewhere, maybe when `SlideshowEdit` unmounts. But that may be overkill.

Reply:
Good thought, thank you. I looked into this more, and from what I understand it would be overkill based on page refresh that should clean this up so the cache is per session, and that the data is lightweight (only image id and size information, not remaining image information). And lastly in attempting to add this it was turning into a complex operation as well.
As for page refresh cleaning the data up, I could confirm this by adding `window.debugMemoCache = memoCache;` to the code, then logging this in the browser console before adding a block with images, and then after a page refresh. After the page refresh it defaults back to a size of 1. And if I remove the Slideshow ad refresh, the cache is gone.
So based on that I think most likely overkill here, would you agree?

Reply:
Absolutely! Thanks for looking it up.